docs/okeps/okep-5088-evpn.md

```diff
@@ -0,0 +1,789 @@
+on the host, as well as configuration on the upstream PE router. Now in order to route Northbound, the pod sends the packet
+to its default gateway, the transit router, which will either route towards ovn-k8s-mpx (local gw mode) or the gateway
+router (GR).
+
+For Layer 3 networks, in OVN-Kubernetes we break up a supernet into a per-node subnet for the PUDN. Whenever a pod on
+a node wants to talk to a pod on another node or anything externally, pure routing is used. This is the equivalent of an
+IP-VRF.
+
+Furthermore, so far we have talked about carrying VPN for Layer 2 and Layer 3 PUDNs. What about the Cluster Default Network (CDN)?
+This network lives in the default VRF. Typically, the default VRF is not carried over the EVPN fabric as a VPN. For
+the purpose of this enhancement it will be considered not be part of an EVPN. The CDN may still use Geneve, or it may
+rely on the upcoming no-overlay mode to use a pure BGP routing scheme.
+
+As far as underlying technologies go, we use FRR as our BGP provider. As EVPN relies on BGP, we will continue to use FRR
+for EVPN support.
+
+### Transitioning to EVPN from Geneve
+
+For Layer 2 and Layer 3 PUDNs Geneve is no longer used when enabling EVPN. Instead, VXLAN is used as an overlay to carry
+per network VPN packets across the EVPN fabric. In practice this means we still need a switch domain that crosses all nodes
+for Layer 2, and we need per node routing for Layer 3. FRR integrates directly with Linux to provide EVPN support. It 
+uses netdevs like Linux Bridge, VXLAN VTEPs, VRFs in order to map Linux networks to advertised EVPN VNIs.
+
+We know that we create a VRF per Layer 2/Layer 3 UDN, which gives us our mapping to an EVPN VRF in Linux that FRR will
+leverage. Therefore, we can conclude that every Layer 2 and Layer 3 UDN maps to an IP-VRF already. This implicit design
+choice precludes us from taking several Layer 2 PUDNs and mapping them to a single IP-VRF. In other words, each MAC-VRF
+in OVN-Kubernetes will map 1 to 1 with an IP-VRF. In practice, the consequence of this is a user is unable to connect
+multiple Layer 2 PUDNs to the same "router" and allow routing between them. However, as we will see later, this can be
+provided by other forms of route leaking (route target importing/exporting) between IP-VRFs. Nonetheless, it is key to
+point out this implicit design choice and why it exists. It also provides implicit Network Isolation which was one of the
+key tenets to the UDN design in the first place. So one could view this as intended behavior.
+
+Additionally, we have the ovn-k8s-mpx interface, which gives us a way to route for Layer 2 networks into the
+VRF. This interface can be viewed as our SVI. It is a way that we can get the packet to the VRF (IP-VRF) to be routed.
+Note the SVI can be the same on every single node, since we are using symmetrical Integrated Routing and Bridging (IRB).
```

Comment:
OK, I was a bit confused here for a sec - I thought this was conflicting with the "non goals", but the actual non-goal is to support **asymmetric** Integrated Routing and Bridging (IRB) with EVPN. 

Dropping this here in case someone else misses the distinction.